### DISCUSSION TITLE
[French morphologizer] Mislabelisation of Mood=Imp|Number=Sing|Tense=Present

### DISCUSSION THREAD

**Discussion by LedaguenelArthur** [explosion/spaCy#8147] [New Features & Project Ideas]
# Mislabelisation of Mood=Imp|Number=Sing|Tense=Present

Hello,

I am using spaCy as part of an ungoing project on French textbooks and I have noticed that verbs found in Imperative, with Singular Number and in Present Tense (ie: "Mange ton repas", "Fais tes devoirs", etc.) are almost systematiccally misclassified as NOUN.

Am I the first one to notice this or is this a well-known biasi of the french pre-trained models ?

After investigating the dataset Sequoia, I think this is because every example of Imperative verbs are given in Plural Number (ie: "Mangez votre repas", "Faites vos devoirs", etc.).
Moreover, Imperative verbs in present tense and singular number are often homographs of NOUNs which makes the task even more difficult (ie: "Forme", "Danse", "Place", etc.)

Anyway, I know this is probably a specific issue to the kind of texts I'm manipulating, that heavily rely on Singular Present Imperative verbs to express the instructions of exercices, and hence a difficult one to adress in a pre-trained model, but I thought it was worth mentioning.

_Idea_: one idea to explore to fix this would be to do data augmentation from the instances in Plural Present Imperative form.

## How to reproduce the behaviour
```
import spacy
nlp = spacy.load("fr_core_news_sm")
doc = nlp("Entoure les verbes au présent.")

for token in doc:
    print(token.text, token.pos_, token.morph)
```

## Info about spaCy

- **spaCy version:** 3.0.6
- **Platform:** Windows-10-10.0.19041-SP0
- **Python version:** 3.7.5
- **Pipelines:** fr_core_news_sm (3.0.0)

**Comment by adrianeboyd** [comment]
Yes, this is due to a mismatch between the training data and your task.

We've thought about doing data augmentation for things like imperatives, questions, and informal forms, which are often missing from the training corpora, but we don't have anything published at this point.

**Comment by LedaguenelArthur** [comment]
Hello,

First of all thank you very much for the quick answer!

I've already looked at several (including Stanza for instance) and it seems that this is a regular blind spot of datasets used to train NLP models. 
By any change are you aware of any POS tagging tool that performs correct labelisation on this kind of data ?

If I were to develop my own instances regarding this specific use case would there be a simple way to _retrain_ the model using those ?

Thanks again,
Best regards,
Arthur Ledaguenel

**Comment by adrianeboyd** [comment]
Let me move this to the discussion board. This issue will be locked, but you can follow the link to the new discussion thread.

**Comment by thjbdvlt** [comment]
In case anyone find this discussion while searching for alternative models for french, I've made a complete pipeline with a `morphologizer` that should usually makes better predictions for Moods/Tenses and for Pronouns -- because the default models haven't seen a lot of personal pronouns, especially "tu".
It's available [here](https://github.com/thjbdvlt/solipCysme). The `parser` isn't very accurate, but the morphologizer should be (99.3 for part-of-speech and 99.2  for morphology -- most errors being adjective/noun inversions in cases like "les jeunes bourgeois·es" or adjective/passive participe verbs).
